src/cloud-api-adaptor/docker/README.md

```diff
@@ -0,0 +1,162 @@
+# Introduction
+
+The `docker` provider simulates a pod VM inside a docker container.
+
+> **Note**: Run the following commands from the root of this repository.
+
+## Prerequisites
+
+- Install `docker` [engine](https://docs.docker.com/engine/install/)
+
+  Ensure docker engine supports API version 1.44+. You can verify it by running 
+  `docker version`.
+  Docker engine version 26+ supports API 1.44.
+
+  Ensure you complete the [post install steps](https://docs.docker.com/engine/install/linux-postinstall/) if using non-root user
+  
+  
+- Kubernetes cluster
+
+## Build CAA pod-VM image
+
+- Set environment variables
+
+```bash
+export CLOUD_PROVIDER=docker
+```
+
+- Build the required pod VM binaries
+ 
+```bash
+cd src/cloud-api-adaptor/docker/image
+make
+```
+
+This will build the required binaries inside a container and place 
+it under `resources/binaries-tree`
+
+- Build the pod VM image
+
+```bash
+make image
+```
+
+This will build the podvm docker image. By default the image is named `quay.io/confidential-containers/podvm-docker-image`.
+
+For quick changes you can just build the binaries of podvm components and update `./resources/binaries-tree/usr/local/bin` with the new
+components and run `make image` to build a new podvm image.
+
+
+You can download a ready-to-use image on your host.
```

Comment:
Maybe we need add a note here, that run the docker pull command on the worker node, where the CAA POD will run on it.